int-set/src/lib.rs

```diff
@@ -86,6 +102,32 @@ impl<T> IntSet<T> {
         IntSet(Membership::Exclusive(BitSet::empty()))
     }
 
+    /// Returns an iterator over all members of the set.
+    /// Note: iteration of inverted sets can be extremely slow due to the very large number of members in the set
+    /// care should be taken when using .iter() in combination with an inverted set.
+    pub fn iter(&self) -> impl Iterator<Item = u32> + '_ {
+        match &self.0 {
+            Membership::Inclusive(s) => int_set_iter(s.iter(), false),
+            Membership::Exclusive(s) => int_set_iter(s.iter(), true),
+        }
+    }
+
+    /// If this is an inclusive membership set then returns an iterator over the members, otherwise returns None.
+    pub fn inclusive_iter(&self) -> Option<impl Iterator<Item = u32> + '_> {
+        match &self.0 {
+            Membership::Inclusive(s) => Some(s.iter()),
+            Membership::Exclusive(_) => None,
+        }
+    }
```

Comment:
not worth changing, but personally I don't think I would add this to public API until I had really felt like I needed it somewhere?